capnpc-go/capnpc-go_test.go

```diff
@@ -467,3 +472,144 @@ func nodeListString(n []*node) string {
 	b.WriteByte(']')
 	return b.String()
 }
+
+func setupTempDir() (string, error) {
+	dir, err := os.MkdirTemp("", "capnpc-go_test")
+	if err != nil {
+		return "", fmt.Errorf("setupTempDir(capnpc-go_test): %v", err)
+	}
+
+	// Add go.mod to the new dir, minimal contents
+	err = os.WriteFile(filepath.Join(dir, "go.mod"),
+		[]byte("module capnpc_go_test"), 0660)
+	if err != nil {
+		return "", fmt.Errorf("setupTempDir: write go.mod: %v", err)
+	}
+
+	// Add go.work to the new dir, minimal contents
+	modRoot, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("setupTempDir: Getwd: %v", err)
+	}
+	modRoot = filepath.Dir(modRoot)
+
+	err = os.WriteFile(filepath.Join(dir, "go.work"),
+		[]byte(fmt.Sprintf("use %s", modRoot)), 0660)
+	if err != nil {
+		return "", fmt.Errorf("setupTempDir: write go.work: %v", err)
+	}
+	return dir, nil
+}
+
+// This test arises because capnproto-c++ has a file named:
+// * src/capnp/persistent.capnp
+// * Also found in this repo: std/capnp/persistent.capnp
+//
+// It contains two definitions:
+//   interface Persistent {}
+//   annotation persistent(interface, field) :Void;
+//
+// testdata/persistent-simple.capnp is a minimal reproducible test case for the
+// collision.
+func TestPersistent(t *testing.T) {
```

Comment:
If possible/reasonable, I would recommend running `t.Parallel()` here.  We have a fairly large test suite, and the parallelism really helps.

Reply:
Good idea, I added t.Parallle() at `capnpac-go_test.go:525`